pkg/apis/pipeline/v1beta1/taskrun_validation.go

```diff
@@ -138,6 +140,24 @@ func validateStepOverrides(overrides []TaskRunStepOverride) (errs *apis.FieldErr
 	return errs
 }
 
+// validateStepOverridesResourceRequirements validates if both step-level and task-level resource requirements are configured
+func validateStepOverridesResourceRequirements(computeResources corev1.ResourceRequirements, overrides []TaskRunStepOverride) (errs *apis.FieldError) {
```

Comment:
apis.ErrMultipleOneOf is probably more appropriate here

Also, I'd suggest the following function name + docstring:

// validateTaskRunComputeResources ensures that compute resources are not configured at both the step level and the task level